contracts/tokenfunds.js

```diff
@@ -0,0 +1,583 @@
+/* eslint-disable no-await-in-loop */
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable max-len */
+/* global actions, api */
+
+const FeeMethod = ['burn', 'issuer'];
+const PayoutType = ['user', 'contract'];
+
+actions.createSSC = async () => {
+  const tableExists = await api.db.tableExists('funds');
+  if (tableExists === false) {
+    await api.db.createTable('funds', ['id', 'lastTickTime']);
+    await api.db.createTable('proposals', ['fundId', 'approvalWeight']);
+    await api.db.createTable('approvals', ['from', 'to']);
+    await api.db.createTable('accounts', ['account']);
+    await api.db.createTable('params');
+
+    const params = {};
+    params.dtfCreationFee = '1000';
+    params.dtfUpdateFee = '300';
+    params.dtfTickHours = '24';
+    params.maxDtfsPerBlock = 40;
+    params.processQueryLimit = 1000;
+    await api.db.insert('params', params);
+  }
+};
+
+actions.updateParams = async (payload) => {
+  const {
+    dtfCreationFee,
+    dtfUpdateFee,
+    dtfTickHours,
+    maxDtfsPerBlock,
+    processQueryLimit,
+  } = payload;
+
+  if (api.sender !== api.owner) return;
+  const params = await api.db.findOne('params', {});
+  if (dtfCreationFee) {
+    if (!api.assert(typeof dtfCreationFee === 'string' && !api.BigNumber(dtfCreationFee).isNaN() && api.BigNumber(dtfCreationFee).gte(0), 'invalid dtfCreationFee')) return;
+    params.dtfCreationFee = dtfCreationFee;
+  }
+  if (dtfUpdateFee) {
+    if (!api.assert(typeof dtfUpdateFee === 'string' && !api.BigNumber(dtfUpdateFee).isNaN() && api.BigNumber(dtfUpdateFee).gte(0), 'invalid dtfUpdateFee')) return;
+    params.dtfUpdateFee = dtfUpdateFee;
+  }
+  if (dtfTickHours) {
+    if (!api.assert(typeof dtfTickHours === 'string' && api.BigNumber(dtfTickHours).isInteger() && api.BigNumber(dtfTickHours).gte(1), 'invalid dtfTickHours')) return;
+    params.dtfTickHours = dtfTickHours;
+  }
+  if (maxDtfsPerBlock) {
+    if (!api.assert(typeof maxDtfsPerBlock === 'string' && api.BigNumber(maxDtfsPerBlock).isInteger() && api.BigNumber(maxDtfsPerBlock).gte(1), 'invalid maxDtfsPerBlock')) return;
+    params.maxDtfsPerBlock = api.BigNumber(maxDtfsPerBlock).toNumber();
+  }
+  if (processQueryLimit) {
+    if (!api.assert(typeof processQueryLimit === 'string' && api.BigNumber(processQueryLimit).isInteger() && api.BigNumber(processQueryLimit).gte(1), 'invalid processQueryLimit')) return;
+    params.processQueryLimit = api.BigNumber(processQueryLimit).toNumber();
+  }
+  await api.db.update('params', params);
+};
+
+function generateDtfId(dtf) {
+  return `${dtf.payToken.replace('.', '-')}:${dtf.voteToken.replace('.', '-')}`;
```

Comment:
What is the purpose of this function?  Why do you need to replace '.' with '-' ?